library/Vanilla/Permissions.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * @copyright 2009-2016 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace Vanilla;
+
+class Permissions {
+
+    /** @var array */
+    private $permissions = [];
+
+    /** @var bool */
+    private $isAdmin = false;
+
+    /** @var bool */
+    private $isBanned = false;
+
+    /**
+     * Permissions constructor.
+     *
+     * @param array $permissions
+     */
+    public function __construct($permissions = []) {
+        $this->setPermissions($permissions);
+    }
+
+    /**
+     * Add a permission.
+     *
+     * @param string $permission
+     * @param int|array $IDs
```

Comment:
IDs of what?

Let's step up the context on some of these doc blocks, especially when we're paving new ground.